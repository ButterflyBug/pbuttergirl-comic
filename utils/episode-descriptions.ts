export const altTexts = [
  [
    "Digital art. The comic shows a girl, 'Peanut butter girl', who has shoulder-length brown hair. The hair is straight and loose with a part in the middle of the head. The girl has brown eyes. Her face is facing a tall brown ledge that extends to her neck. On the shelf is a jar completely filled with peanut butter. The lid of this jar is red and the jar says 'Peanut Butter'. There is a picture of a peanut in a shell between the words 'Peanut' and 'butter'. The girl is smiling, staring at the jar and thinking, 'Aww ... we meet again, m'lady'. ",
  ],
  [
    "Digital art. The picture shows two friends standing next to each other. The girl on the left is slightly turned to the girl on the right. The girl on the left has blonde hair, green eyes, and is wearing a black blouse with small white polka dots. The girl on the right has brown hair, brown eyes, and is wearing a black blouse with thin straps and a plunging neckline. The girls are ready to go out to the party. The friends are smiling as they are in the process of taking another selfie before going out. Both girls have the same thought about this selfie - 'This time will be perfect ...'. In the lower left corner of the comic there is the inscription 'Selfie # 10244'",
  ],
  [
    "Digital art. The picture shows a girl sitting in a room with purple walls on a purple floor cross-legged in a meditation position. The girl has brown hair and a bun pinned up high. Her eyes are closed. The girl looks very focused. She is wearing a black tank top, dark purple leggings and black socks. At the height of her head, green shapes of electromagnetic waves can be seen on both sides. Next to it is a small brown cabinet with four drawers. There is a navy blue WIFI router on the cabinet, which also sends electromagnetic waves in navy blue. In the upper left corner of the picture there is a speech bubble saying: 'I will transfer my thougths to him, so that I will not have to tell about them'. On the right side of the wall there is a piece of paper with a red inscription 'No Hard Feelings'.",
  ],
  ['Forth episode', 'Forth episode 2', 'Forth episode 3'],
  [
    "Digital art. The picture shows a girl wearing a yellow long-sleeved blouse and black leggings. Her hair is brown in a ponytail. The eyes are also brown. The girl is wearing orange headphones. The girl is dancing. Her left arm is stretched out in front of her and her right arm is placed on the waist. The background of the picture is blue. In the upper right corner there is a speech bubble with the word 'Dance' and its phonetic transcription. The flag of Great Britain is drawn in the lower right corner.",
  ],
];
