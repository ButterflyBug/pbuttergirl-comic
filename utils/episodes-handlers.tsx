import glob from "glob";
import { groupBy } from "lodash";

export type EpisodeType = ReturnType<typeof getEpisodes>[0];

export function getEpisodes() {
  const episodes = glob.sync("./episodes/episode-*/*.png");
  const groupedList = groupBy(episodes, groupFunction);
  return Object.entries(groupedList).map(([key, value]) => {
    return {
      name: (key.charAt(0).toUpperCase() + key.slice(1)).replace("-", " "),
      images: value,
    };
  });
}

export const groupFunction = (episodePath: string) => {
  return episodePath.split("/")[2];
};
