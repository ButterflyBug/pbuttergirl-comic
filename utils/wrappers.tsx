import initHeadManager from 'next/dist/client/head-manager';
import { HeadManagerContext } from 'next/dist/shared/lib/head-manager-context';
import { useEffect } from 'react';

export const HeadWrapper: React.FC = props => {
  const { children } = props;
  const manager = initHeadManager();

  useEffect(() => {
    const meta = document.createElement('meta');
    meta.content = '0';
    meta.name = 'next-head-count';
    document.getElementsByTagName('head')[0].appendChild(meta);
  });

  return (
    <HeadManagerContext.Provider value={manager}>
      {children}
    </HeadManagerContext.Provider>
  );
};
