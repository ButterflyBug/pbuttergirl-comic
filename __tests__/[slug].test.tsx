/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import EpisodePage from '../pages/episodes/[slug]';
import { getEpisodes } from '../utils/episodes-handlers';
import { Container } from 'react-bootstrap';

describe('EpisodePage', () => {
  it('renders a heading', () => {
    const episode = getEpisodes()[0];
    const episodes = getEpisodes();
    render(
      <EpisodePage
        episode={episode}
        currentEpisodeNumber={1}
        episodes={episodes}
      />
    );

    const heading = screen.getByRole('heading', {
      name: /Episode 1/i,
    });

    const image = screen.getByTestId('comic-image');

    expect(heading).toBeInTheDocument();
    expect(image).toBeInTheDocument();
  });

  it('renders a title', async () => {
    const episode = getEpisodes()[0];
    const episodes = getEpisodes();
    const { container } = render(
      <EpisodePage
        episode={episode}
        currentEpisodeNumber={1}
        episodes={episodes}
      />
    );

    await waitFor(() => {
      expect(container.querySelector('head')).toMatchSnapshot();
    });
  });
});
