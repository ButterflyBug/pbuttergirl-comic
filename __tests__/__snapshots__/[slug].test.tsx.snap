// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodePage renders a title 1`] = `null`;
